### DISCUSSION TITLE
Proper interpretation of "depth" from `magick identify -verbose`

### DISCUSSION THREAD

**Discussion by gregbenz** [ImageMagick/ImageMagick#5487] [General] [answered]
When a file reports "Depth 10/16-Bit", what does that mean?

Is that saying the true image data is 10-bits, but saved in a 16-bit format? Something else? The nomenclature is unclear to me.

**Comment by snibgo** [comment]
When we have two numbers, the first is metadata for the image (which can be changed with "`-depth N`"), and the second is actually calculated from the image by finding the number of bits needed to record each pixel value.

**Reply by gregbenz** [reply]
If the metadata is corrupt, is there a way to confirm the actual bit depth as encoded?
Or are you saying that the 16 in this case indicates the test file (an AVIF) was encoded at a higher bit depth than expected?

**Comment by snibgo** [comment]
> ... is there a way to confirm the actual bit depth as encoded?

As encoded where? In memory, IM encodes the image according to the Q-number and whether IM is built with HDRI. For a given build of IM, all images are encoded in memory in the same way.

If you mean encoded in a file, then IM tells you nothing about that (although the first number *may* be the bit depth from an input file). If you want that metadata, use exiftool.

**Comment by gregbenz** [comment]
Yes, I mean was the image encoded with 10 or 12-bits (converting HDR images to JXL and AVIF through a couple of tools). Ultimately, looking to confirm that data was written with the expected quality.

Is there not a way with IM to check the bit depth of the encoded image as saved in the file?

**Comment by snibgo** [comment, marked as answer]
IM tells you nothing about that (although the first number may be the bit depth from an input file). If you want that metadata, use exiftool.

**Comment by gregbenz** [comment]
Interesting, would have assumed this could be pulled from the data. Thank you for your help!
